android/library/src/main/java/com/liferay/mobile/screens/viewsets/defaultviews/ddm/form/fields/DDMFieldGridView.kt

```diff
@@ -0,0 +1,178 @@
+
+import android.content.Context
+import android.graphics.Typeface
+import android.support.design.widget.Snackbar
+import android.support.v7.widget.TooltipCompat
+import android.util.AttributeSet
+import android.view.LayoutInflater
+import android.view.View
+import android.widget.LinearLayout
+import android.widget.TextView
+import com.liferay.mobile.screens.R
+import com.liferay.mobile.screens.ddl.form.view.DDLFieldViewModel
+import com.liferay.mobile.screens.ddl.model.Option
+import com.liferay.mobile.screens.ddm.form.model.Grid
+import com.liferay.mobile.screens.ddm.form.model.GridField
+import com.liferay.mobile.screens.ddm.form.model.get
+import com.liferay.mobile.screens.thingscreenlet.delegates.bindNonNull
+import com.liferay.mobile.screens.viewsets.defaultviews.util.ThemeUtil
+import org.jetbrains.anko.childrenSequence
+import rx.Observable
+import rx.Subscriber
+import rx.Subscription
+
+/**
+ * @author Victor Oliveira
+ */
+open class DDMFieldGridView @JvmOverloads constructor(context: Context, attrs: AttributeSet? = null,
+	defStyleAttr: Int = 0) : LinearLayout(context, attrs, defStyleAttr), DDLFieldViewModel<GridField> {
+	private lateinit var gridField: GridField
+	private lateinit var parentView: View
+	private val labelTextView: TextView by bindNonNull(R.id.liferay_ddm_label)
+	private val hintTextView: TextView by bindNonNull(R.id.liferay_ddm_hint)
+	val gridLinearLayout: LinearLayout by bindNonNull(R.id.liferay_ddm_grid)
+
+	private var changeValuesSubscription: Subscription? = null
+	private var changeValuesSubscriber: Subscriber<in Boolean>? = null
+	private var changeValuesGridSubscriber: Subscriber<in GridField>? = null
+	private val changeValuesObservable = Observable.create<Boolean> { changeValuesSubscriber = it }
+
+	override fun getField(): GridField {
+		return this.gridField
+	}
+
+	override fun setField(field: GridField) {
+		this.gridField = field
+
+		setupLabelLayout()
+		setupFieldLayout()
+	}
+
+	override fun getOnChangedValueObservable(): Observable<GridField> {
+		return Observable.create<GridField> {
+			changeValuesGridSubscriber = it
+		}
+	}
+
+	private fun onColumnValueChanged(which: Int, row: Option, ddmFieldGridRowView: DDMFieldGridRowView) {
+		val option = this.gridField.columns[which]
+
+		if (this.gridField.currentValue == null) {
+			this.gridField.currentValue = Grid(mutableMapOf(row.value to option.value))
+		} else {
+			this.gridField.currentValue.rawValues[row.value] = option.value
+		}
+
+		val columnEditText = ddmFieldGridRowView.columnSelectView.textEditText
+		columnEditText.setTypeface(columnEditText.typeface, Typeface.BOLD)
+
+		changeValuesSubscriber?.onNext(field.isValid)
+		changeValuesGridSubscriber?.onNext(field)
+	}
+
+	private fun setupFieldLayout() {
+		val inflater = LayoutInflater.from(context)
+		val layoutIdentifier = ThemeUtil.getLayoutIdentifier(context, "ddmfield_grid_row")
+
+		this.gridField.rows.forEach { row ->
```

Comment:
inconsistent use of this :P

Reply:
Truth, I'll fix this in the next commits.